### PR TITLE
Yili - Clicking active button cause the page to freeze 

### PR DESCRIPTION
**Pull request by suaniii**
# Description
Clicking active button cause the page to freeze

## Related PRS (if any):
For backend use development branch

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as owner/ admin user
5. go to dashboard→ Reports→ Reports →People
6. Hover over any green dot or grey dot within the "Active" section
7. Confirm that the cursor remains as an arrow (not a clickable pointer), aligning with the behavior on the "Project" and "Teams" pages.
8. Click Test: Click on any green dot or grey dot.
9. Confirm that the page does not freeze and remains responsive.

## Screenshots or videos of changes:

https://github.com/user-attachments/assets/a6e107f1-02a6-445b-a245-d05138b16b1a

### CODE REVIEW

**Review by vijay10pk — DISMISSED** [review]
Changes are working as expected.

https://github.com/user-attachments/assets/e5be7589-280f-43ba-9a19-1100e2a6efbc

**Review by CodewithKoushica — CHANGES_REQUESTED** [review]
Everything works as expected! One thing to consider: is the onStatusChange functionality necessary here? Please let me know your thoughts.

https://github.com/user-attachments/assets/d59d2c67-02f3-4bea-93e7-b909a82457c7

**Comment by suaniii** [comment]
> Everything works as expected! One thing to consider: is the onStatusChange functionality necessary here? Please let me know your thoughts.
> 
>  Screen.Recording.2024-10-29.at.10.04.11.AM.mov

Thank you for the feedback! I don’t believe the onStatusChange functionality is necessary in this context. Firstly, the active status on this page is determined by the "Start Date" and "End Date" fields, so there's no need for manual status changes. Secondly, this is a Reports page, not a User Management page, making onStatusChange redundant here. We already have the status change functionality implemented in the User Management page (as shown in the image), where it’s more appropriate and relevant.

Screen Shot 2024-10-29 at 1 22 28 PM

File: src/components/Reports/PeopleTable.js [hunk 1]
```diff
@@ -6,6 +6,10 @@ import { boxStyle, boxStyleDark } from 'styles';
 
 function PeopleTable({ userProfiles, darkMode }) {
   let PeopleList = [];
+  function onStatusClick(person, isActive) {
```
[the remark this answers is not in the document]
Comment (samarth9201) [reply src/components/Reports/PeopleTable.js:9]: I agree, should be removed

**Review by samarth9201 — DISMISSED** [review]
Changes working as expected.

Screenshot 2024-10-29 154031
Screenshot 2024-10-29 154040

**Comment by CodewithKoushica** [comment]
> > Everything works as expected! One thing to consider: is the onStatusChange functionality necessary here? Please let me know your thoughts.
> > Screen.Recording.2024-10-29.at.10.04.11.AM.mov
> 
> Thank you for the feedback! I don’t believe the onStatusChange functionality is necessary in this context. Firstly, the active status on this page is determined by the "Start Date" and "End Date" fields, so there's no need for manual status changes. Secondly, this is a Reports page, not a User Management page, making onStatusChange redundant here. We already have the status change functionality implemented in the User Management page (as shown in the image), where it’s more appropriate and relevant.
> 
> Screen Shot 2024-10-29 at 1 22 28 PM

Thank you for clarifying! That makes complete sense—if the active status is fully controlled by the "Start Date" and "End Date" fields here, then onStatusChange does seem unnecessary. Could you please remove the onStatusChange functionality from this page?

@suaniii  Thanks again for the thoughtful explanation!

**Review by anirudhsk2107 — DISMISSED** [review]
The changes are working properly 

Screen Shot 2024-10-29 at 11 45 35 PM

**Review by hqzhu0913 — CHANGES_REQUESTED** [review]
Function works as expected. And like others mentioned, onStatusClick is not used anyway and status should not be clickable.
Screenshot 2024-10-29 at 11 23 02 PM
Screenshot 2024-10-29 at 11 39 12 PM

**Comment by beblicarl** [comment]
This functionality works as intended and the code is great

[6ebd4b8b-681e-460f-9702-c5f894d1508a.webm](https://github.com/user-attachments/assets/3c1f2c07-66cc-42e4-ae43-19776cccdb20)

**Review by beblicarl — APPROVED** [review]

**Review by nikhilgiri0226 — DISMISSED** [review]
Hi Yili,

I have reviewed the PR and screen was not freeze after clicking the green/grey dot. There was one slight UI related concern. When I clicked on the People table, the table seems to be very wide and table for teams and projects were as expected. I tested this in edge, chrome and I saw this behavior in both browser. Maybe this is at my end only so I am not suggesting any changes because other reviewers were not facing the same issue. So, approving this change.

Kindly find the video below for reference.


https://github.com/user-attachments/assets/f6c50a9f-dfcd-4ad9-b2b4-7aa2c714b657

**Review by Jingyii800 — DISMISSED** [review]
The feature worked well. The page doesn't freeze when clicking on the active button. But the total layout for People looks weird on my screen.

https://github.com/user-attachments/assets/dc15d764-2d9a-4411-a7d4-073c558e04d7

**Review by ajitp4292 — APPROVED** [review]
Page does not freeze, works has expected.
Screenshot 2024-11-02 at 8 01 57 PM

**Review by varun-elango — COMMENTED** [review]
It is working as expected. The green/grey dots are not clickable and the page does not freeze. I do agree with others that the onStatusClick function is not needed. And the page is rendered improperly and I had to zoom out a lot to see the green dots. But that is not in scope of this PR.
PR2823_1
PR2823_2

**Comment by suaniii** [comment]
Edited. Please check it again.
Thanks for the comments.

**Comment by CodewithKoushica** [comment]
Everything is working as expected. Thank you for considering my request for the change.
Approving the PR.

**Review by CodewithKoushica — APPROVED** [review]

**Review by hqzhu0913 — COMMENTED** [review]
Hi, I test all people, team and project's active button. They are not clickable and work as expected.

Just one thing, I am not sure if this is my problem. The people table overflow in the test. 

https://www.loom.com/share/cd11220c176646c6bdee3babce564e25?sid=ef1eeb43-6a3e-45b7-87f6-25f33e0fa4de

**Review by Muhideenthegreat — APPROVED** [review]
Verification:

- Page no longer freezes upon click of the dots and cursor remains as an arrow over the dots as well. Bug fixed ! 

Screenshot 2024-11-07 at 4 07 34 PM

**Review by nikhilpittala16 — APPROVED** [review]
I have tested the PR and everything works as expected.

https://github.com/user-attachments/assets/b8261222-d9b5-4b5c-b0e0-40fc2a31175a

**Review by hqzhu0913 — APPROVED** [review]
It works as expected. And no more clickable function.

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**